content/Encapsulation/Encapsulation.q.md

```diff
@@ -0,0 +1,1470 @@
+?>
+```
+/// type=SS, id=91909009-182d-4fcf-9c6a-bd39790e4ac1, answer=[3]
+
+In the `echo` statement on line 41, replace `$personObject->getName()` with `$personObject->name`. Execute the program. What is its output?
+
+ - It prints `Anna`.
+
+ - It prints `John`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `John is 15 years old.`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+class Person
+{
+    private $name = "Anna";
+    private $age;
+
+    public function setName($newName)
+    {
+        $this->name = $newName;
+    }
+
+    public function setAge($newAge)
+    {
+        if ($this->checkAge($newAge)) {
+            $this->age = $newAge;
+        }
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function getAge()
+    {
+        return $this->age;
+    }
+
+    private function checkAge($age)
+    {
+        if ($age > 0) {
+            return true;
+        }
+        return false;
+    }
+}
+
+$personObject = new Person();
+$personObject->setAge(15);
+$personObject->setName("John");
+echo $personObject->name . " is " . $personObject->getAge() . " years old.";
+?>
+```
+/// type=MS, id=acc2cdb9-e555-4a58-8b23-9ea6893c003a, answer=[3,4,5]
+
+Why does accessing the `$name` property of `$personObject` using `$personObject->name` produce an error?
+
+ - There is no `$name` property defined in the `Person` class.
+
+ - `$personObject` is not declared as the instance of the `Person` class.
+
+ - The private property `$name` is not accessible outside of the `Person` class.
+
+ - The `$name` property is encapsulated and accessible only inside the `Person` class.
+
+ - `$personObject` is not allowed to directly access the `$name` property of the `Person` class.
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=MS, id=2f2ccb77-5f86-41a7-ba8f-6ac89835689e, answer=[2,4,5]
+
+Which statements are true about encapsulation?
+
+ - It is a process of creating a new object of a class.
+
+ - It restricts other objects to directly access the properties of a certain class.
+
+ - It is a process of changing the behavior of a parent class method in a child class.
+
+ - It is a way of wrapping the properties and methods of a class together as a single unit.
+
+ - It hides the internal implementation of an object from other objects for protection purposes.
+
+
+/// type=MS, id=77231f1f-5b27-4eb5-a6ba-4fa30e695eec, answer=[3,5]
+
+Which statements are the main purpose of encapsulation?
+
+ - It improves code readability.
+
+ - It removes unnecessary methods.
+
+ - It protects the internal state of an object.
+
+ - It shortens the implementation of the code.
+
+ - It reduces software development complexity.
+
+
+/// type=SS, id=dc0e186a-15dc-4745-9fa9-99c86e07c6c2, answer=[2]
+
+Which visibility keyword is used to encapsulate properties and methods in a class?
+
+ - `public`
+
+ - `private`
```

Comment:
this does not encapsulate but limits the accessibility of the property

Reply:
This is noted. Rephrased the question statement already. Thank you.